examples/billing/add_billing_setup.rb

```diff
@@ -0,0 +1,175 @@
+  client,
+  customer_id,
+  payments_account_id,
+  payments_profile_id)
+  billing_setup = client.operation.create_resource.billing_setup do |b|
+    # Sets the appropriate payments account field.
+    if payments_account_id
+      # If a payments account ID has been provided, set the resource name.
+      # You can list available payments accounts via the PaymentsAccountService's
+      # ListPaymentsAccounts method.
+      b.payments_account = client.path.payments_account(customer_id, payments_account_id)
+    elsif payments_profile_id
+      # Otherwise, create a new payments account by setting the payments account info.
+      # See https://support.google.com/google-ads/answer/7268503 for more information about
+      # payments profiles.
+      b.payments_account_info = client.resource.payments_account_info do |pai|
+        pai.payments_account_name = "Payment Account #{(Time.new.to_f * 1000).to_i}"
+        pai.payments_profile_id = payments_profile_id
+      end
+    else
+      raise "No payments account ID or payments profile ID is provided."
+    end
+  end
+  billing_setup
+end
+
+def set_billing_setup_date_times(client, customer_id, billing_setup)
+  # The query to search existing approved billing setups in the end date time
+  # descending order.
+  # See get_billing_setup.rb for a more detailed example of how to retrieve
+  # billing setups.
+  query = <<~QUERY
+    SELECT billing_setup.end_date_time
+    FROM billing_setup
+    WHERE billing_setup.status = APPROVED
+    ORDER BY billing_setup.end_date_time DESC
+  QUERY
+
+  # Issues a search stream request.
+  stream = client.service.google_ads.search_stream(
+    customer_id: customer_id,
+    query: query,
+  )
+
+  start_date = DateTime.parse(Date.today.strftime("%Y%m%d"))
+
+  # Retrieves the ending date time of the last billing setup.
+  row = stream.first.results.first
+  if row
+    puts row
+    last_ending_date_time_string = row.billing_setup.end_date_time
+    if last_ending_date_time_string.nil?
+      raise "Cannot set starting and ending date times for the new billing setup." \
+        "The latest existing billing setup is set to run indefinitely."
+    end
+    # Sets the new billing setup to start one day after the ending date time.
+    start_date = DateTime.parse(last_ending_date_time_string) + 1
+  end
+
+  # Sets the new billing setup start date and end date.
+  billing_setup.start_date_time = start_date.strftime("%Y-%m-%d")
+  billing_setup.end_date_time = (start_date + 1).strftime("%Y-%m-%d")
+end
+
+if __FILE__ == $0
+  options = {}
+  # The following parameter(s) should be provided to run the example. You can
+  # either specify these by changing the INSERT_XXX_ID_HERE values below, or on
+  # the command line.
+  #
+  # Parameters passed on the command line will override any parameters set in
+  # code.
+  #
+  # Running the example with -h will print the command line usage.
+  options[:customer_id] = 'INSERT_CUSTOMER_ID_HERE'
```

Comment:
Set up a space for users to put in default values for the other arguments up here as well.

Reply:
Done